backpack-compose/src/main/kotlin/net/skyscanner/backpack/compose/progressbar/BpkProgressBar.kt

```diff
@@ -0,0 +1,122 @@
+/**
+ * Backpack for Android - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package net.skyscanner.backpack.compose.progressbar
+
+import androidx.compose.animation.core.animateFloatAsState
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.Arrangement
+import androidx.compose.foundation.layout.Box
+import androidx.compose.foundation.layout.Row
+import androidx.compose.foundation.layout.fillMaxHeight
+import androidx.compose.foundation.layout.fillMaxWidth
+import androidx.compose.foundation.layout.height
+import androidx.compose.foundation.layout.width
+import androidx.compose.foundation.progressSemantics
+import androidx.compose.foundation.shape.CircleShape
+import androidx.compose.material3.LinearProgressIndicator
+import androidx.compose.material3.ProgressIndicatorDefaults
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.draw.clip
+import androidx.compose.ui.graphics.StrokeCap
+import net.skyscanner.backpack.compose.theme.BpkTheme
+import androidx.compose.ui.unit.Dp
+import net.skyscanner.backpack.compose.tokens.BpkBorderSize
+import net.skyscanner.backpack.compose.tokens.BpkSpacing
+
+enum class BpkProgressBarSize {
+    Small,
+    Large,
+}
+
+@Composable
+fun BpkProgressBar(
+    value: Float,
+    modifier: Modifier = Modifier,
+    size: BpkProgressBarSize = BpkProgressBarSize.Small,
+    min: Int = 0,
+    max: Int = 1,
+    stepped: Boolean = false,
+) {
+    val height = if (size == BpkProgressBarSize.Small) BpkSpacing.Md else BpkSpacing.Base
+
+    val progress = value
+        .coerceIn(min.toFloat(), max.toFloat())
+        .div(max.toFloat())
+
+    val animatedProgress by animateFloatAsState(
+        targetValue = progress,
+        animationSpec = ProgressIndicatorDefaults.ProgressAnimationSpec,
+    )
+
+    Box(modifier = modifier) {
+
+        val numberOfSteps = max
+
+        LinearProgressIndicator(
+            progress = animatedProgress,
+            modifier = Modifier
+                .height(height)
+                .fillMaxWidth()
+                .clip(CircleShape)
+                .progressSemantics(
```

Comment:
can you also include the values range? that should match the min/max values (that's mainly why they exist in the api)

Reply:
I left it out because we convert `progress` to a float between 0..1 and coerce it [here](https://github.com/Skyscanner/backpack-android/pull/1658/files/61eceb48942db0db87039c2313242bfb98ce640d#diff-4f9bd9b6cdd30c54e45a5dabb963a30753c5d7bb8a1547788135f320933c02ffR60-R62). As the default `valueRange` parameter is 0..1 we should be able to use that.

Reply:
yeah that's what i meant in my original comment with setting the raw value and the range - that way we'll announce the actual value within the min/max range

Reply:
Would that be as useful for someone using the screen reader? For example, in the following it will say "0% progress" when it's loaded 50% of the progress which gives off the wrong impression of how much of the data is loaded.

<img width="344" alt="Screenshot 2023-07-21 at 09 21 46" src="https://github.com/Skyscanner/backpack-android/assets/26006896/7572ed54-077a-4eb7-80ac-89c787ec93a4">

Reply:
I asked @olliecurtis for help, so he will probably come back to you as I'm also not 100% sure what the right approach is here

Reply:
Ok I had a chat with @olliecurtis. We think we can remove min from the API/component as it doesn't make a lot of sense. We should still set the value & valueRange based on max here, but min doesn't make a lot of sense and it should always go from 0 (as it does visually).

Reply:
I agree. If we're going to change the API, I think it's worth making the following changes:

### Remove `max` and state the value will always be clamped between 0..100

This removes the overhead of "If a parse max in I need to parse a value in between 0..max and not 0..1.", which will be required if users want to add steps to the component.

### Change `value: Float` to `percentage: Int`

When the progress is an integer (this is true for [flights](StaggeredProgressBarView) and hotels - we receive `"completionPercentage": Int` in the DTO), users have to convert to a `Float`, which looks like it will be most of the time.

### Change `stepped (Boolean)` to `numberOfSteps (Int)` and set it default to 1

Asking "How many steps are there to the progress?" with a single variable is easier to understand than "The number of steps will be max - 1, and by default max is 1 so you will not see any steps even if you set `stepped == true` if you do not set max != 1".

So will end up like this:

```
@Composable
fun BpkProgressBar(
    percentage: Int, // value between 0..100
    modifier: Modifier = Modifier,
    size: BpkProgressBarSize = BpkProgressBarSize.Small,
    numberOfSteps: Int = 1, // 1..100
)
```

Wdyt?

Reply:
Had another chat with @olliecurtis and it still makes sense to support the max value for both the stepped and non-stepped version as you may have some extra information about the maximum value. for example of web we get back x providers out of y and the progress bar indicates that. 
i think having `numberOfSteps` combined with `percentage` can be equally confusing as you need to do mapping between the percentage and step count. also you may have 55.5% for example and an int may not be enough.

so to keep things in sync across platforms we will need to maintain this (min will be removed on web too).

Reply:
Fair enough, done